homeassistant/components/unifiprotect/sensor.py

```diff
@@ -0,0 +1,407 @@
+"""This component provides sensors for UniFi Protect."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+import logging
+from typing import Any, Callable, Sequence
+
+from pyunifiprotect.data import NVR
+from pyunifiprotect.data.base import ProtectAdoptableDeviceModel
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    DATA_BYTES,
+    DATA_RATE_BYTES_PER_SECOND,
+    DATA_RATE_MEGABITS_PER_SECOND,
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_ILLUMINANCE,
+    DEVICE_CLASS_SIGNAL_STRENGTH,
+    DEVICE_CLASS_TEMPERATURE,
+    DEVICE_CLASS_TIMESTAMP,
+    DEVICE_CLASS_VOLTAGE,
+    ELECTRIC_POTENTIAL_VOLT,
+    PERCENTAGE,
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    TEMP_CELSIUS,
+    TIME_SECONDS,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import Entity, EntityCategory
+
+from .const import DOMAIN
+from .data import ProtectData
+from .entity import ProtectDeviceEntity, ProtectNVREntity, async_all_device_entities
+from .models import ProtectRequiredKeysMixin
+from .utils import get_nested_attr
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class ProtectSensorEntityDescription(ProtectRequiredKeysMixin, SensorEntityDescription):
+    """Describes UniFi Protect Sensor entity."""
+
+    precision: int | None = None
+
+
+_KEY_MEMORY = "memory_utilization"
+_KEY_DISK = "storage_utilization"
+_KEY_RECORD_ROTATE = "record_rotating"
+_KEY_RECORD_TIMELAPSE = "record_timelapse"
+_KEY_RECORD_DETECTIONS = "record_detections"
+_KEY_RES_HD = "resolution_HD"
+_KEY_RES_4K = "resolution_4K"
+_KEY_RES_FREE = "resolution_free"
+_KEY_CAPACITY = "record_capacity"
+
+ALL_DEVICES_SENSORS: tuple[ProtectSensorEntityDescription, ...] = (
+    ProtectSensorEntityDescription(
+        key="uptime",
+        name="Uptime",
+        icon="mdi:clock",
+        device_class=DEVICE_CLASS_TIMESTAMP,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
```

Comment:
Why is this one disabled? Does it update frequently?

Reply:
Most of the diagnostic sensors are disabled by default. I do not have the docs in front of me, but I remember there was something in the integration docs that said if the sensor was not commonly used, it should be disabled by default. 

Uptime is probably safe to enable though (and not it does not update often, it is a fixed point in time like it is suppose to be).

Many of the other ones that do update often though (CPU Utilization for example) also requires a configuration option enabled for them to update in real-time. By default, we are filtering out a ton of Websocket messages to reduce CPU usage.

Reply:
So, entity categories have been introduced to prevent cluttering (which was one of the reasons to disable by default previously); so for those cases, we don't need to disable them anymore.

Another reason to disable by default, is for quickly updating sensors (like CPU utilization indeed, or signal strength and such).

Looking at the PR, the sensors are disabled by common use and not the above reasoning.

Reply:
Okay, so probably a fair way to do it then is any sensor that expects the Real-time metrics config option should be disabled by default? Otherwise enable them.

Reply:
Yeah, and even real-time, could be enabled if a primary entity of course (like.. dunno temperature).

Reply:
Enabled them all except for 

* Camera: received/transferred data (this might also have a lot of overlap with the UniFi Network integration
* NVR: CPU utilization, temperature and memory utilization (both require real time to be of any use)

Many of the others are _less_ useful with real-time not on, but still useful.